code/_onclick/item_attack.dm

```diff
@@ -71,7 +71,7 @@
 		return
 
 	if(obj_flags & CAN_BE_HIT)
-		return I.attack_obj(src, user)
+		return attacking_item.attack_obj(src, user)
```

Comment:
I think you're complicating the proc chain here, although thats probabably because it was bad in the first place.
attackby calls attack_obj here, which in turn calls the extremely well named attacked_by, where the actual damage is done.
Rather than this confusing shit, all the bonus damage stuff should be handed in attacked_by.

Then all you really need to do is call attacking_item.attack_obj(src, user) again if we don't already have a return value (and checking CAN_BE_HIT again I guess)